homeassistant/components/here_travel_time/coordinator.py

```diff
@@ -113,7 +114,7 @@ def _parse_routing_response(self, response) -> HERETravelTimeData:
         destination_name: str | None = None
         if (names := section["spans"][-1].get("names")) is not None:
             destination_name = names[0]["value"]
-        return HERETravelTimeData(
+        return HERETravelTimeData(  # type: ignore[no-any-return]
```

Comment:
This implies that `HERETravelTimeData` is not typed.
I think we do not allow strict-typing if the underlying library is not typed.

Reply:
Thank you! If this is the case I will close the PR. The underlying lib is indeed not (yet) typed.

`HERETravelTimeData` is a TypedDict defined in [model.py](https://github.com/home-assistant/core/blob/dev/homeassistant/components/here_travel_time/model.py#L9) I am puzzled why mypy has a problem here.

Reply:
That's weird. Let me check...

Reply:
OK, so I think you have two issues here.
In HERETravelTimeData definition, you have used the capital letter definitions. I think you meant to use the value of the coresponding constant:
```python
class HERETravelTimeData(TypedDict):
    """Routing information."""

    attribution: str | None
    duration: float
    duration_in_traffic: float
    distance: float
    origin: str
    destination: str
    origin_name: str
    destination_name: str
```

The second issue is that you are building HERETravelTimeData from a dict. You should instead use init from kwargs:
```python
        return HERETravelTimeData(
            ATTR_ATTRIBUTION=None,
            ATTR_DURATION=round(summary["baseDuration"] / 60),  # type: ignore[misc]
            ATTR_DURATION_IN_TRAFFIC=round(summary["duration"] / 60),
            ATTR_DISTANCE=distance,
            ATTR_ORIGIN=f"{mapped_origin_lat},{mapped_origin_lon}",
            ATTR_DESTINATION=f"{mapped_destination_lat},{mapped_destination_lon}",
            ATTR_ORIGIN_NAME=origin_name,
            ATTR_DESTINATION_NAME=destination_name,
        )
```

Reply:
Thank you!!!